HIS/his_checker.py

```diff
@@ -0,0 +1,177 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) Bao Project and Contributors. All rights reserved
+
+import os
+import sys
+import subprocess
+
+cyclomatic_treshold = 10
+comment_density = 0.2
+goto_treshold = 0
+call_treshold = 7
+callee_treshold = 5
+statements_treshold = 50
+
+def process_metrics(filename):
+
+    """Process each HIS metric."""
+
+    process_complexity(filename)
+    process_comments(filename)
+    process_statements(filename)
+    process_callees(filename)
+    process_goto(filename)
+    process_calls(filename)
+
+def process_complexity(file):
+
+    """Process McCabe Cyclomatic Complexity for each function in a file"""
+
+    complexity = "pmccabe -c "
+    lines = os.popen(complexity + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        colonyCount = int(fields[0])
+        if colonyCount > cyclomatic_treshold:
+            eprint("Complexity exceeded (max." + str(cyclomatic_treshold) + "): "
+                    + fields[0] + " at " + fields[5])
+            sys.exit(-1)
+
+def process_comments(file):
+
+    """Process comment density in a file"""
+
+    tokei = "tokei -f "
+    lines = os.popen(tokei + str(file))
+    lines = lines.read()
+    sline = lines.split('\n')
+
+    for num in range(len(sline)-1):
+        fields = sline[num].split('\t')
+        fields = sline[num].strip()
+        i=1
+        param = []
+        files = -1
+        lines = -1
+        code = -1
+        comments = -1
+        j = 0
+        if (str(fields[0]).startswith("/")):
+            fields = fields.split(' ')
+            while i<len(fields):
+                if (fields[i].isdigit()):
+                    if (files == -1): files = int(fields[i])
+                    elif (lines == -1): lines = int(fields[i])
+                    elif (code == -1): code = float(fields[i])
+                    elif (comments == -1): comments = float(fields[i])
+                    if (comments > -1 and code > 0 and (comments/code)<comment_density
+                            and fields[0].endswith(".c")): 
+                        print("Comment density smaller than " + str(comment_density)
+                                 + ": "  + str(comments/code)+"  at "+fields[0])
```

Comment:
There is no need to add '(' after the if statement. Also, using 1 statement in each line helps with code reading (lines 66-69).
```suggestion
        if str(fields[0]).startswith("/"):
            fields = fields.split(' ')
            while i < len(fields):
                if fields[i].isdigit():
                    if files == -1:
                        files = int(fields[i])
                    elif lines == -1:
                        lines = int(fields[i])
                    elif code == -1:
                        code = float(fields[i])
                    elif comments == -1:
                        comments = float(fields[i])
                    if (comments > -1 and code > 0 and (comments/code) < COMMENT_DENSITY
                            and fields[0].endswith(".c")):
                        print("Comment density smaller than " + str(COMMENT_DENSITY) +
                              ": "  + str(comments/code)+"  at "+fields[0])
```

Reply:
Removed '(' and changed statements. Thx